### PR TITLE
CEDS-2400 EIDR pages

### PR DESCRIPTION
**Pull request by MaciejRewera**
This PR adds a sub-journey related to Entry Into Declarant's Records.
As part of this there are also changes to Navigator class that allows to
have back link dependent on the cached data.

### CODE REVIEW

File: app/controllers/navigation/Navigator.scala [hunk 1]
```diff
@@ -273,6 +333,28 @@ object Navigator {
         common.orElse(specific)(page)(mode)
     }
 
+  def backLink(page: DeclarationPage, mode: Mode, cacheModel: ExportsDeclaration)(implicit request: JourneyRequest[_]): Call =
```
Comment (PatrykRudnicki) [inline app/controllers/navigation/Navigator.scala:336]: As I mentioned yesterday `cacheModel` is in the request. You're taking this from the same request to pass to the method.

Comment (PatrykRudnicki) [reply app/controllers/navigation/Navigator.scala:336]: WIthout this change, you don't need to add this cache model as argument anywhere

File: app/controllers/navigation/Navigator.scala [hunk 2]
```diff
@@ -255,6 +266,55 @@ object Navigator {
...
+      controllers.declaration.routes.RepresentativeStatusController.displayPage(mode)
+
+  private def consignorEoriNumberClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call = {
+    val declarantEori: Option[Eori] = cacheModel.parties.declarantDetails.flatMap(_.details.eori)
+    val exporterEori: Option[Eori] = cacheModel.parties.exporterDetails.flatMap(_.details.eori)
+
+    if (declarantEori.isDefined && exporterEori.isDefined && declarantEori == exporterEori)
+      controllers.declaration.routes.DeclarantExporterController.displayPage(mode)
+    else
+      controllers.declaration.routes.ExporterDetailsController.displayPage(mode)
```
Comment (PatrykRudnicki) [inline app/controllers/navigation/Navigator.scala:309]: This logic should be based on the answer Are you Exporter

Comment (MaciejRewera) [reply app/controllers/navigation/Navigator.scala:309]: I am not the person to address this comment as I have only copied what was already there.
Seems we have a bug then.

Comment (PatrykRudnicki) [reply app/controllers/navigation/Navigator.scala:309]: It has been fixed in https://github.com/hmrc/customs-declare-exports-frontend/pull/899/files

Comment (PatrykRudnicki) [reply app/controllers/navigation/Navigator.scala:309]: I'll change it during rebasing

File: app/views/components/gds/Styles.scala [hunk 3]
```diff
@@ -19,6 +19,8 @@ package views.components.gds
 object Styles {
 
   val gdsPageLegend = "govuk-fieldset__legend--l"
+  val gdsPageLegendLarge = "govuk-fieldset__legend--xl"
```
Comment (SteveSugden) [inline app/views/components/gds/Styles.scala:22]: Why are we adding "large" styles?   The plan was to change the definitions from "...l" to "...xl" at some point

Comment (MaciejRewera) [reply app/views/components/gds/Styles.scala:22]: Wasn't aware of that. '--l' did not match the prototype so I created those.
So should it be just '--l' ?

Comment (SteveSugden) [reply app/views/components/gds/Styles.scala:22]: Yes.  That is the reason for creating the constants - so we can switch to the "larger" styles en-mass when all pages are GDS

File: app/controllers/navigation/Navigator.scala [hunk 4]
```diff
@@ -255,6 +265,55 @@ object Navigator {
...
+      controllers.declaration.routes.DeclarantExporterController.displayPage(mode)
+
+  private def carrierDetailsClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call =
+    if (cacheModel.parties.declarantIsExporter.exists(_.isExporter))
+      controllers.declaration.routes.DeclarantExporterController.displayPage(mode)
+    else
+      controllers.declaration.routes.RepresentativeStatusController.displayPage(mode)
+
+  private def consigneeDetailsSupplementaryPreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call =
+    if (cacheModel.`type` == SUPPLEMENTARY && cacheModel.parties.declarantIsExporter.exists(_.isExporter))
```
Comment (SteveSugden) [inline app/controllers/navigation/Navigator.scala:296]: This is already a "SUPPLEMENTARY" method so no need to check again?

**Comment by MaciejRewera** [comment]
There are some major changes to be done before merging (rebasing onto other branches at least) so I will open a new PR with that.

**Pull request closed without merging**